addons/binding/org.openhab.binding.nibeuplink/src/main/java/org/openhab/binding/nibeuplink/handler/GenericUplinkHandler.java

```diff
@@ -0,0 +1,219 @@
+ *
+ */
+public abstract class GenericUplinkHandler extends BaseThingHandler implements NibeUplinkHandler {
+
+    private final String NO_VALUE = "--";
+
+    private final Logger logger = LoggerFactory.getLogger(GenericUplinkHandler.class);
+
+    private Set<Channel> deadChannels = new HashSet<>(1000);
+
+    /**
+     * Refresh interval which is used to poll values from the NibeUplink web interface (optional, defaults to 60 s)
+     */
+    private int refreshInterval;
+
+    /**
+     * Refresh interval which is used clean the dead channel list (optional, defaults to 1 h)
+     */
+    private int houseKeepingInterval = 1;
+
+    /**
+     * Interface object for querying the FRITZ!Box web interface
+     */
+    private UplinkWebInterface webInterface;
+
+    /**
+     * Job which will do the FRITZ!Box polling
+     */
+    private final UplinkPolling pollingRunnable;
+
+    /**
+     * Schedule for polling
+     */
+    private ScheduledFuture<?> pollingJob;
+
+    /**
+     * Schedule for periodic cleaning dead channel list
+     */
+    private ScheduledFuture<?> deadChannelHouseKeeping;
+
+    public GenericUplinkHandler(@NonNull Thing thing) {
+        super(thing);
+        this.pollingRunnable = new UplinkPolling(this);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("command for {}: {}", channelUID, command);
+
+        // TODO: necessary??? will be read only in the first place
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("About to initialize NibeUplink");
+        Thing thing = this.getThing();
+        NibeUplinkConfiguration config = getConfiguration();
+
+        logger.debug("Discovered NibeUplink initialized: {}", config);
+
+        this.refreshInterval = config.getPollingInterval();
+        this.houseKeepingInterval = config.getHouseKeepingInterval();
+        this.webInterface = new UplinkWebInterface(config, this);
+
+        if (config.getPassword() != null) {
+            this.startPolling();
+        } else {
+            thing.setStatusInfo(
+                    new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "no password set"));
+        }
+    }
+
+    /**
+     * Start the polling.
+     */
+    private synchronized void startPolling() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            logger.debug("start polling job at intervall {}", refreshInterval);
+            pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 1, refreshInterval, TimeUnit.SECONDS);
+        } else {
+            logger.debug("pollingJob already active");
+        }
+        if (deadChannelHouseKeeping == null || deadChannelHouseKeeping.isCancelled()) {
+            logger.debug("start deadChannelHouseKeeping job at intervall {}", houseKeepingInterval);
+            deadChannelHouseKeeping = scheduler.scheduleWithFixedDelay(new Runnable() {
```

Comment:
This can (and is preferred as java 8 lambda notation:   `scheduler.scheduleWithFixedDelay(deadChannels::clear, ...`

Reply:
fixed it